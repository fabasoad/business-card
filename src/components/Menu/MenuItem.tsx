import * as React from 'react'
import { Nav } from 'react-bootstrap'
import { useTranslation } from 'react-i18next'

interface MenuItemProps {
  name: string
  setActiveNavLink: React.Dispatch<React.SetStateAction<string>>
}

export default function MenuItem(props: MenuItemProps) {
  const { t } = useTranslation()
  return (
    <Nav.Link
      onClick={() => props.setActiveNavLink(`#${props.name}`)}
      eventKey={`#${props.name}`}
      href={`#${props.name}`}
    >
      {t(`business-card-menu-${props.name}`)}
    </Nav.Link>
  )
}
