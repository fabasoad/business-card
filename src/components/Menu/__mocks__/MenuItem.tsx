import * as React from 'react'
import type { MenuItemProps } from '../MenuItem'

export default function MenuItem({ name, setActiveNavLink }: MenuItemProps) {
  return (
    <div data-testid={name}>{setActiveNavLink.name}</div>
  )
}
