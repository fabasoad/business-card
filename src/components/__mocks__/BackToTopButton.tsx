import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
