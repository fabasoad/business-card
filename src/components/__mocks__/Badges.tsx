import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
