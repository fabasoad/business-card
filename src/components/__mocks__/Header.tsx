import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
