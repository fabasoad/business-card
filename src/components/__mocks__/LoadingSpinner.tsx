import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
