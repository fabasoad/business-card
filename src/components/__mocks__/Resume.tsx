import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
