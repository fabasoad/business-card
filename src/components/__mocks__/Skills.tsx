import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
