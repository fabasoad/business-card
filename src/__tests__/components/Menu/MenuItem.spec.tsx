/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import MenuItem from '../../../components/Menu/MenuItem'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock;

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render MenuItem correctly', () => {
  const name = 'test-menu-item'
  const wrapper: ShallowWrapper = shallow(<MenuItem
    name={name}
    setActiveNavLink={null}
  />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(1)
  expect(tMock).toHaveBeenCalledWith(`business-card-menu-${name}`)
})

test('should set active MenuItem correctly', () => {
  const name = 'test-menu-item'
  const setActiveNavLinkSpy = jest.fn()
  const wrapper: ShallowWrapper = shallow(<MenuItem
    name={name}
    setActiveNavLink={setActiveNavLinkSpy}
  />)
  wrapper.find('NavLink').simulate('click')
  expect(setActiveNavLinkSpy).toHaveBeenCalledWith(`#${name}`)
})
