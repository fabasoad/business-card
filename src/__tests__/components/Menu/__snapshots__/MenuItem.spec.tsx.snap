// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MenuItem correctly 1`] = `
<NavLink
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  disabled={false}
  eventKey="#test-menu-item"
  href="#test-menu-item"
  onClick={[Function]}
>
  business-card-menu-test-menu-item
</NavLink>
`;
